### PR TITLE
refactoring: ChangefeedInfo struct

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
1. remove `Status` field in `ChangefeedInfo`
2. remove some function about `filter` of `ChangefeedInfo` and add struct `TxnFilter`
3. Adjustment the responsibility of `owner` and `changfeed`

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: cdc/owner.go [hunk 1]
```diff
@@ -580,73 +587,102 @@ func (c *changeFeed) pullDDLJob() error {
...
-			log.Debug("update changefeed", zap.String("id", cf.ID),
-				zap.Uint64("checkpoint ts", minCheckpointTs),
-				zap.Uint64("resolved ts", minResolvedTs))
+// handleDDL call handleDDL of every changefeeds
+func (o *ownerImpl) handleDDL(ctx context.Context) error {
+	for _, cf := range o.changeFeeds {
+		err := cf.handleDDL(ctx, o.captures)
+		switch errors.Cause(err) {
+		case nil:
+			// do nothing, continue
```
Comment (suzaku) [inline cdc/owner.go]: Why not just `continue`?

File: cdc/owner_test.go [hunk 2]
```diff
@@ -347,10 +353,10 @@ func (s *ownerSuite) TestDDL(c *check.C) {
...
 	cfID := "test_handle_admin"
 	sampleCF := &changeFeed{
-		ID:               cfID,
-		info:             &model.ChangeFeedInfo{},
-		ChangeFeedStatus: &model.ChangeFeedStatus{},
-		State:            model.ChangeFeedSyncDML,
+		ID:       cfID,
+		info:     &model.ChangeFeedInfo{},
+		status:   &model.ChangeFeedStatus{},
+		DDLState: model.ChangeFeedSyncDML,
```
Comment (suzaku) [inline cdc/owner_test.go]: It seems `DDLState` is not a suitable name.

Comment (leoppro) [reply cdc/owner_test.go]: what suggestion do you have?

File: cdc/scheduler.go [hunk 3]
```diff
@@ -278,11 +287,19 @@ func realRunProcessorWatcher(
...
-	sw := NewProcessorWatcher(changefeedID, captureID, pdEndpoints, etcdCli, info)
+) (*ProcessorWatcher, error) {
+	status, err := kv.GetChangeFeedStatus(ctx, etcdCli, changefeedID)
+	if err != nil && errors.Cause(err) != model.ErrChangeFeedNotExists {
+		return nil, errors.Trace(err)
+	}
+	checkpointTs := info.GetStartTs()
+	if status != nil {
+		checkpointTs = status.CheckpointTs
+	}
```
Comment (amyangfei) [inline cdc/scheduler.go]: extract a simple function for this logic, can be reused in cdc/owner.go#L551-554

**Comment by leoppro** [comment]
PTAL again @amyangfei

**Pull request merged**